File: bindings/recipes/python/02_verify_authority.py
from __future__ import annotations

import json
import os
from pathlib import Path

from auths.verify import verify


root = Path(os.environ["AUTHS_RECIPE_FIXTURE"])
proof = (root / "workflow.proof.cbor").read_bytes()
action = (root / "workflow.action.cbor").read_bytes()
context = (root / "workflow.context.cbor").read_bytes()
verified = verify(proof, action, context)
if verified.kind != "authorized":
    raise RuntimeError(f"unexpected verdict: {verified.kind}")
changed = bytearray(action)
changed[-1] ^= 1
try:
    changed_rejected = verify(proof, bytes(changed), context).kind != "authorized"
except (TypeError, ValueError):
    changed_rejected = True
if not changed_rejected:
    raise RuntimeError("mutated action remained authorized")
print(
    json.dumps(
        {
            "recipe": "02-verify-authority",
            "outcome": verified.kind,
            "changedRejected": changed_rejected,
        }
    )
)

File: bindings/recipes/typescript/02-verify-authority.ts
import { readFile } from "node:fs/promises";
import { createVerifier } from "@auths-dev/sdk/verify";

const fixture = process.env.AUTHS_RECIPE_FIXTURE;
if (fixture === undefined) throw new Error("AUTHS_RECIPE_FIXTURE is required");
const [proof, action, trustedContext] = await Promise.all([
  readFile(`${fixture}/workflow.proof.cbor`),
  readFile(`${fixture}/workflow.action.cbor`),
  readFile(`${fixture}/workflow.context.cbor`),
]);
const verifier = await createVerifier();
const result = verifier.verify({ proof: new Uint8Array(proof), action: new Uint8Array(action), trustedContext: new Uint8Array(trustedContext) });
if (result.kind !== "authorized") throw new Error(result.code);
console.log(JSON.stringify({ recipe: "02-verify-authority", outcome: result.kind }));
